aws/sdk-codegen/src/test/kotlin/software/amazon/smithy/rustsdk/EndpointsCredentialsTest.kt

```diff
@@ -0,0 +1,145 @@
+                          "conditions": [{"fn": "isSet", "argv": [{"ref": "AuthMode"}]}],
+                          "endpoint": {
+                            "url": "https://example.com"
+                            "properties": {
+                                "authSchemes": [{
+                                    "name": "sigv4",
+                                    "signingRegion": "region-{AuthMode}",
+                                    "signingName": "name-{AuthMode}"
+                                }]
+                            }
+                          }
+                      }],
+            "parameters": {
+                "AuthMode": { "required": false, "type": "String" },
+                "Region": { "required": false, "type": "String", "builtIn": "AWS::Region" }
+            }
+        })
+        @restJson1
+        service FooBaz {
+            version: "2018-03-17",
+            operations: [DefaultAuth, CustomAuth]
+        }
+
+        @http(uri: "/default", method: "GET")
+        @staticContextParams({ AuthMode: { value: "default-auth" } })
+        operation DefaultAuth { }
+
+
+        @http(uri: "/custom", method: "GET")
+        @staticContextParams({ AuthMode: { value: "custom-auth" } })
+        operation CustomAuth { }
+    """.asSmithyModel()
+
+    @Test
+    fun `endpoint rules configure auth in default and non-default case`() {
+        clientIntegrationTest(
+            model, runtimeConfig = AwsTestRuntimeConfig,
+            additionalSettings = ObjectNode.builder()
+                .withMember(
+                    "customizationConfig",
+                    ObjectNode.builder()
+                        .withMember(
+                            "awsSdk",
+                            ObjectNode.builder()
+                                .withMember("integrationTestPath", "../sdk/integration-tests")
+                                .build(),
+                        ).build(),
+                )
+                .withMember("codegen", ObjectNode.builder().withMember("includeFluentClient", false).build()).build(),
+        ) { context, rustCrate ->
+            val moduleName = context.moduleUseName()
+            rustCrate.integrationTest("endpoints_auth") {
+                // assert that a rule with no default auth works properly
+                tokioTest("default_auth") {
+                    rustTemplate(
+                        """
+                        let (conn, rcvr) = #{capture_request}(None);
+                        let conf = $moduleName::Config::builder()
+                            .http_connector(conn)
+                            .region(#{Region}::new("us-west-2"))
+                            .credentials_provider($moduleName::Credentials::new("example", "example", None, None, "example"))
+                            .build();
+                        let client = $moduleName::Client::from_conf(conf);
+                        let _ = dbg!(client.default_auth().send().await);
```

Comment:
Should we prefer using `dbg!` in tests? Why not just unwrap here?

Reply:
ah I think it was me debugging why it wasn't working